src/effection/child_process.ts

```diff
@@ -17,41 +17,42 @@ function supervise(execution: Execution, child: ChildProcess) {
   // process.
   //
   // More information here: https://unix.stackexchange.com/questions/14815/process-descendants
-  execution.atExit(() => {
+  execution.ensure(() => {
     try {
       process.kill(-child.pid, "SIGTERM")
     } catch(e) {
       // do nothing, process is probably already dead
     }
   });
-  execution.monitor(function*() {
+
+  execution.spawn(monitor(function*() {
     let [error]: [Error] = yield on(child, "error");
     throw error;
-  });
+  }));
 
-  execution.monitor(function*() {
+  execution.spawn(monitor(function*() {
     let [code]: [number] = yield on(child, "exit");
     if(code !== 0) { throw new Error("child exited with non-zero exit code") }
-  });
+  }));
 }
 
-export function spawn(command: string, args?: ReadonlyArray<string>, options?: SpawnOptions): Controller {
+export function spawn(command: string, args?: ReadonlyArray<string>, options?: SpawnOptions): Operation {
   return (execution) => {
     let child = childProcess.spawn(command, args, Object.assign({}, options, {
       shell: true,
       detached: true,
     }));
-    supervise(execution, child);
+    supervise(execution.context.parent, child);
     execution.resume(child);
   }
 }
 
-export function fork(module: string, args?: ReadonlyArray<string>, options?: ForkOptions): Controller {
+export function fork(module: string, args?: ReadonlyArray<string>, options?: ForkOptions): Operation {
   return (execution) => {
     let child = childProcess.fork(module, args, Object.assign({}, options, {
       detached: true,
     }));
-    supervise(execution, child);
+    supervise(execution.context.parent, child);
```

Comment:
There are two changes to be aware of 

1. The signature of what gets passed into a controller function is a `Control` object https://github.com/thefrontside/effection.js/blob/ea4c1ca00fabea0194cb2b97e13d16b3e06a721a/index.d.ts#L20-L27 not the actual context
2. Whereas before the control function ran _inside the context from which it was invoked_. It now gets its own clean environment to run in. If you want to effect the environment from which you were invoked, you need to use `context.parent`